### PR TITLE
Autogeneration of documentation

### LINKED ISSUE #18: Static typing with mypy

### PR DESCRIPTION
**Pull request by robertodr**
Implements the autogeneration of input documentation. This is the last bit of the functionality we had planned for the library :confetti_ball: 
- For the moment, the only format implemented is reStructuredText. Given that one can use Sphinx to generate HTML or PDF, or even use Pandoc to do more fanciful conversions, I don't see any compelling reasons to implement any other format.
- Flake8 and mypy checks can be run within pytest (passing `--flake8 parselglossy` and `--mypy parselglossy`, respectively) This is now in the Travis set up. Fix #18 
- Fix up and test the API functions. 
- Fix up and test CLI.

And this is how it looks:
Screenshot from 2019-03-25 14-58-34

### CODE REVIEW

**Comment by bast** [comment]
I agree to not redo what Pandoc can do and to restrict to RST.

**Comment by robertodr** [comment]
CLI tests fail on Windows only. Seems it can read back the files.

File: README.rst [hunk 1]
```diff
@@ -39,6 +39,10 @@ parselglossy
 
    -- **1 Corinthians** 14:18
 
+   Parse it on!
+
+   -- Bobson Dugnutt, **Private communication**
```
Comment (bast) [inline README.rst:44]: hahaha

**Review by bast — APPROVED** [review]
Ready to merge once CI is happy. Thank you!

File: README.rst (hunk 1 above)
Comment (robertodr) [reply README.rst:44]: :octopus:

**Pull request merged**